model/migration.go

```diff
@@ -4,6 +4,7 @@
 package model
 
 const (
+	ADVANCED_PERMISSIONS_MIGRATION_KEY         = "AdvancedPermissionsMigrationComplete"
```

Comment:
Why are you moving this to model? It's not part of the migrations model but a special "migration" that exists entirely in the app layer.

Reply:
Also has nothing to do with telemetry.

Reply:
This is used as part of the telemetry here https://github.com/mattermost/mattermost-server/pull/14832/files#diff-6d075395bdd93d9b78959b73d7a2f3d9R931 so we have 2 options, or use directly the string in the telemetry to get key, or expose this outside of the app to avoid a cyclic dependency.

Other option is to remove that data from telemetry (Probably we have some data that we can remove from telemetry because is no longer interesting).

Reply:
Gross, but 0/5 and not really your fault - it's just a somewhat inevitable side effect of having that app layer migration in the first place...